torch_tvm/custom_tvm_ops/cpp/relay/custom_dense_init.cc

```diff
@@ -0,0 +1,37 @@
+#include <relay/op/op_common.h>
+#include <relay/op/type_relations.h>
+#include <relay/pass/alter_op_layout.h>
+#include <tvm/expr.h>
+#include <tvm/operation.h>
+#include <tvm/relay/op.h>
+#include "custom_dense.h"
+
+namespace tvm {
+namespace relay {
+
+RELAY_REGISTER_OP("nn.custom_dense")
```

Comment:
Why do we need this, instead of alterlayout?

Reply:
Here is a code example: 

https://github.com/ajtulloch/tvm/blob/sparse-ops/topi/python/topi/x86/dense.py#L264-L295

Reply:
Yeah, sorry, I meant that we don't need to explicit add a packing op.

Reply:
Yeah, it doesn't have any restriction on nchw and stuff. It takes an op, add packing logic and return a new op.

Reply:
Yeah, exactly, it' done by alter_op_layout relay optimization pass, which hooks into the altert_op_layout function.